### PR TITLE
[5.1] Allow Collection argument in HasOneOrMany::saveMany

### PR DESCRIPTION
**Pull request by ryanwinchester**
When working with Eloquent models we usually have a single Model or a Collection of Models.

It makes sense to allow us to use Collections...

### CODE REVIEW

File: src/Illuminate/Database/Eloquent/Relations/HasOneOrMany.php [hunk 1]
```diff
@@ -217,14 +217,18 @@ public function save(Model $model)
     /**
      * Attach an array of models to the parent instance.
      *
-     * @param  array  $models
+     * @param  array|Collection $models
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Relations/HasOneOrMany.php]: Please don't delete that extra space.

File: src/Illuminate/Database/Eloquent/Relations/HasOneOrMany.php [hunk 2]
```diff
@@ -217,14 +217,18 @@ public function save(Model $model)
...
      * @return array
      */
-    public function saveMany(array $models)
+    public function saveMany($models)
     {
-        array_walk($models, [$this, 'save']);
+        if (! $models instanceof Collection) {
```
Comment (lucasmichot) [inline src/Illuminate/Database/Eloquent/Relations/HasOneOrMany.php]: Why not just 
```php
return (new Collection($models))->each(function (Model $model) {
    $this->save($model);
});
```

File: src/Illuminate/Database/Eloquent/Relations/HasOneOrMany.php (hunk 1 above)
Comment (lucasmichot) [inline src/Illuminate/Database/Eloquent/Relations/HasOneOrMany.php]: Should be either `mixed` either `array|\Illuminate\Database\Eloquent\Collection`
This is not a `\Illuminate\Support\Collection`

Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Relations/HasOneOrMany.php]: > This is not a \Illuminate\Support\Collection

It technically is. :P

File: src/Illuminate/Database/Eloquent/Relations/HasOneOrMany.php (hunk 2 above)
Comment (fungku) [inline src/Illuminate/Database/Eloquent/Relations/HasOneOrMany.php]: Ah, that would work too, I guess.

Comment (fungku) [inline src/Illuminate/Database/Eloquent/Relations/HasOneOrMany.php]: No performance concerns with this?

**Pull request merged**